acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/acceptance/plugins/RpcEndpointServicePluginTest.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.tests.acceptance.plugins;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.hyperledger.besu.config.JsonUtil;
+import org.hyperledger.besu.tests.acceptance.dsl.AcceptanceTestBase;
+import org.hyperledger.besu.tests.acceptance.dsl.node.BesuNode;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import org.junit.Before;
+import org.junit.Test;
+
+public class RpcEndpointServicePluginTest extends AcceptanceTestBase {
+
+  private BesuNode node;
+
+  private OkHttpClient client;
+  protected static final MediaType JSON = MediaType.parse("application/json; charset=utf-8");
+
+  @Before
+  public void setUp() throws Exception {
+    node = besu.createPluginsNode("node1", List.of("testPlugins"), List.of("--rpc-http-api=TESTS"));
+    cluster.start(node);
+    client = new OkHttpClient();
+  }
+
+  @Test
+  public void canUseRpcToSetValue() throws IOException {
+    String setValue = "secondCall";
+
+    ObjectNode resultJson = callTestMethod("tests_getValue", List.of());
+    assertThat(resultJson.get("result").asText()).isEqualTo("InitialValue");
+
+    resultJson = callTestMethod("tests_setValue", List.of(setValue));
+    assertThat(resultJson.get("result").asText()).isEqualTo(setValue);
+
+    resultJson = callTestMethod("tests_getValue", List.of("ignored"));
+    assertThat(resultJson.get("result").asText()).isEqualTo(setValue);
+  }
+
+  @Test
+  public void canCheckArgumentInsideSetValue() throws IOException {
+    ObjectNode resultJson = callTestMethod("tests_setValue", List.of("one", "two"));
+    assertThat(resultJson.get("error").get("message").asText()).isEqualTo("Internal error");
+  }
+
+  @Test
+  public void canThrowExceptions() throws IOException {
+    ObjectNode resultJson = callTestMethod("tests_throwException", List.of());
+    assertThat(resultJson.get("error").get("message").asText()).isEqualTo("Internal error");
```

Comment:
we don't get kaboom?

Reply:
Changed the method name `exceptionsInPluginMethodReturnError` to reflect intent